flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertex.java

```diff
@@ -499,13 +497,12 @@ public void fail(Throwable t) {
         }
     }
 
-    public void resetForNewExecution(final long timestamp, final long expectedGlobalModVersion)
-            throws GlobalModVersionMismatch {
+    public void resetForNewExecution(final long timestamp) {
```

Comment:
it may be out of the scope of this task but I think this method is no longer needed in production code and can be removed.